### DISCUSSION TITLE
Command to remove all unused typescript imports

### DISCUSSION THREAD

**Discussion by alexandersokolow** [neoclide/coc.nvim#4550] [Q&A]
When I run <Plug>(coc-codeaction-cursor) cursor over an unused import in a typescript file, a quickfix is available under the name of "Delete all unused imports", which does what I want.
However, I would like to create a vim binding to trigger this quickfix (if it is available), without having to hover over the list and press the corresponding index.

I already echoed the list of codeActions to a file & found the fixId "unusedIdentifier_deleteImports". Is there some way to apply a fix with this id?

**Comment by jackielii** [comment]
```vim
nnoremap <leader>gi :call CocAction('runCommand', 'editor.action.organizeImport')<CR>
```

**Reply by alexandersokolow** [reply]
Thanks for the answer, but this doesn't do what I need. It also organizes the remaining imports, whereas I just wanted it to delete the unused ones. There is a separate action for that, I already knew about the organizeImport one, should've mentioned that I guess.

**Reply by jackielii** [reply]
Oh. ok sorry.

In this case, what I usually do is see what "commands" is available for this. Just did a quick search, there isn't one.

Also you could try to turn on server trace, and execute your action and see what response it's sending. If there is one, you can simulate it using CocAction, mostly like via a plugin. Making single file plugin can be useful

**Comment by yigitkalan** [comment]
Hi i have a question, you said when you run (coc-codeaction-cursor), how do you run it. I can't make that work i have lua config and use the remappings in the example configuration but it doesnt work.
This is the line i use :
```
keyset("n", "<leader>av", "<Plug>(coc-codeaction-cursor)", opts)
```

**Reply by alexandersokolow** [reply]
Sorry for the late reply. I'm not sure how to do in in lua, in my vimscript config I am doing the following:
`nnoremap <leader>c <Plug>(coc-codeaction-cursor)`

**Reply by yigitkalan** [reply]
Thanks anyway. I found how to do it :)
